src/components/tldraw/index.js

```diff
@@ -0,0 +1,199 @@
+
+  assets[`slide-background-asset-${id}`] = {
+    id: `slide-background-asset-${id}`,
+    size: [width || 0, height || 0],
+    src: buildFileURL(src),
+    type: "image",
+  };
+
+  shapes["slide-background-shape"] = {
+    assetId: `slide-background-asset-${id}`,
+    childIndex: 0.5,
+    id: "slide-background-shape",
+    name: "Image",
+    type: TDShapeType.Image,
+    parentId: 1,
+    point: [0, 0],
+    isLocked: true,
+    size: [width || 0, height || 0],
+    style: {
+      dash: DashStyle.Draw,
+      size: SizeStyle.Medium,
+      color: ColorStyle.Blue,
+    },
+  };
+
+  if (index === -1 || isEmpty(interval)) return { assets, shapes, width, height }
+
+  for (let i = 0; i < interval.length; i++) {
+    if (!interval[i]) continue;
+
+    const tldrawData = getTldrawData(index);
+
+    const {
+      shape,
+    } = tldrawData[i];
+
+    shapes[shape.id] = shape;
+  }
+
+  return { assets, shapes, width, height }
+}
+
+const getCameraAndZoom = (index) => {
+  const inactive = {
+    xCamera: 0,
+    yCamera: 0,
+    zoom: 0,
+  };
+
+  if (index === -1) return inactive;
+
+  const currentData = storage.panzooms[index];
+
+  return {
+    xCamera: currentData.xCamera,
+    yCamera: currentData.yCamera,
+    zoom: currentData.zoom,
+  };
+};
+
+const TldrawPresentation = ({ size }) => {
+  const [tldrawAPI, setTLDrawAPI] = React.useState(null);
+  const intl = useIntl();
+  const currentContent = useCurrentContent();
+  const currentPanzoomIndex = useCurrentIndex(storage.panzooms);
+  const currentSlideIndex = useCurrentIndex(storage.slides);
+  const started = currentPanzoomIndex !== -1;
+
+  const result = SlideData();
+
+  let { assets, shapes } = result;
+
+  const calculateCameraFitSlide = (bounds, result) => {
+    const { width, height } = result;
+    if (bounds && width && height) {
+      let zoom = 
+        Math.min(
+          (bounds.width) / width,
+          (bounds.height) / height,
+        )        
+        
+      zoom = Utils.clamp(zoom, 0.1, 5);
+    
+      let point = [0, 0];
+      if ((bounds.width / bounds.height) > 
+          (width / height)) 
+      { 
+        point[0] = (bounds.width - (width * zoom)) / 2 / zoom
+      } else {
+        point[1] = (bounds.height - (height * zoom)) / 2 / zoom
+      }
+    
+      return {point, zoom}
+    } else {
+      return null
+    }
+  }
+
+  React.useEffect(() => {
+    const { xCamera, yCamera, zoom } = getCameraAndZoom(currentPanzoomIndex);
+    if (xCamera === 0 && yCamera === 0 && zoom === 0) {
+      const camera = calculateCameraFitSlide(tldrawAPI?.rendererBounds, result);
+      if (camera) {
+        tldrawAPI?.setCamera(camera.point, camera.zoom);
+      }
+    } else {
+      tldrawAPI?.setCamera([xCamera, yCamera], zoom);
+    }
+  }, [currentPanzoomIndex, currentSlideIndex, tldrawAPI, result]);
+  
+  React.useMemo(() => {
+    tldrawAPI?.replacePageContent(shapes, {}, assets)
+  }, [tldrawAPI, shapes, assets]);
```

Comment:
Genuine question here: doesn't the injected `React.memo` used at the bottom's export where `areEqual` is always `true` affects this hook?

Reply:
Hmmm not sure about this one also, will ask to others to take a look.